fendermint/app/src/cmd/upgrade.rs

```diff
@@ -0,0 +1,45 @@
+// Copyright 2022-2024 Protocol Labs
+// SPDX-License-Identifier: Apache-2.0, MIT
+
+use std::path::PathBuf;
+
+use fendermint_app_options::upgrade::AddUpgrade;
+use fendermint_app_options::upgrade::UpgradeArgs;
+use fendermint_app_options::upgrade::UpgradeCommands;
+use fendermint_vm_interpreter::fvm::upgrades::UpgradeSchedule;
+use fendermint_vm_message::ipc::UpgradeInfo;
+
+use crate::cmd;
+
+cmd! {
+    UpgradeArgs(self) {
+        let upgrade_file = self.upgrade_file.clone();
+        match &self.command {
+            UpgradeCommands::AddUpgrade(args) => args.exec(upgrade_file).await,
+        }
+    }
+}
+
+cmd! {
+    AddUpgrade(self, upgrade_file: PathBuf) {
+        let mut us = if upgrade_file.exists() {
+            // load existing upgrade schedule
+            UpgradeSchedule::from_file(&upgrade_file)?
+        } else {
+            // create an empty upgrade schedule
+            UpgradeSchedule::new().to_file(&upgrade_file)?;
+            UpgradeSchedule::new()
+        };
```

Comment:
Maybe a `UpgradeSchedule::load_or_init` or `get_or_create` could be used to encapsulate this? It appears in the same way further up.